modules/transport-netty4/src/main/java/org/elasticsearch/transport/netty4/ESLoggingHandler.java

```diff
@@ -42,92 +30,8 @@
-                    final int length = arg.getInt(offset + MESSAGE_LENGTH_OFFSET);
-                    if (length == TcpTransport.PING_DATA_SIZE) {
-                        sb.append(" [ping]").append(' ').append(eventName).append(": ").append(readableBytes).append('B');
-                        return sb.toString();
-                    }
-                }
-                else if (readableBytes >= TcpHeader.HEADER_SIZE) {
-                    // we are going to try to decode this as an ES message
-                    final int length = arg.getInt(offset + MESSAGE_LENGTH_OFFSET);
-                    final long requestId = arg.getLong(offset + REQUEST_ID_OFFSET);
-                    final byte status = arg.getByte(offset + STATUS_OFFSET);
-                    final boolean isRequest = TransportStatus.isRequest(status);
-                    final String type = isRequest ? "request" : "response";
-                    final String version = Version.fromId(arg.getInt(offset + VERSION_ID_OFFSET)).toString();
-                    sb.append(" [length: ").append(length);
-                    sb.append(", request id: ").append(requestId);
-                    sb.append(", type: ").append(type);
-                    sb.append(", version: ").append(version);
-                    if (isRequest) {
-                        // it looks like an ES request, try to decode the action
-                        final int remaining = readableBytes - ACTION_OFFSET;
-                        final ByteBuf slice = arg.slice(offset + ACTION_OFFSET, remaining);
-                        // the stream might be compressed
-                        try (StreamInput in = in(status, slice, remaining)) {
-                            // the first bytes in the message is the context headers
-                            try (ThreadContext context = new ThreadContext(Settings.EMPTY)) {
-                                context.readHeaders(in);
-                            }
-                            // now we decode the features
-                            if (in.getVersion().onOrAfter(Version.V_6_3_0)) {
-                                in.readStringArray();
-                            }
-                            // now we can decode the action name
-                            sb.append(", action: ").append(in.readString());
-                        }
-                    }
-                    sb.append(']');
-                    sb.append(' ').append(eventName).append(": ").append(readableBytes).append('B');
-                    return sb.toString();
-                }
-            }
-        }
-        // we could not decode this as an ES message, use the default formatting
-        return super.format(ctx, eventName, arg);
+    public void channelReadComplete(ChannelHandlerContext ctx) throws Exception {
+        // We do not want to log read complete events because we log inbound messages in the TcpTransport.
+        ctx.fireChannelReadComplete();
```

Comment:
I am not sure I miss something but I wonder if we still need this handler at all if we don't log anything?